### PR TITLE
fix: (Platform) table enhancements

### PR DESCRIPTION
**Pull request by olshansky**
#### Please provide a link to the associated issue.
Part of https://github.com/sap/fundamental-ngx/issues/4735
#### Please provide a brief summary of this pull request.
- Fixed row selection by Space/Enter button in single & multiple modes
- Fixed `ERROR Error: ExpressionChangedAfterItHasBeenCheckedError: Expression has changed after it was checked.`
- Added ability to scroll table in "page scroll" by keyboard 
- Minor a11y enhancements _(`headers` for cells, `aria-colindex/rowindex`, roles, scopes)_
- Fixed anchors on the examples page  
#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [n/a] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [n/a] update `README.md`
- [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/platform/src/lib/components/table/directives/table-cell-selectable.directive.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { AfterContentChecked, AfterViewChecked, Directive, HostListener } from '@angular/core';
+import { KeyUtil } from '@fundamental-ngx/core';
+import { DOWN_ARROW, PAGE_DOWN, PAGE_UP, SPACE, UP_ARROW } from '@angular/cdk/keycodes';
```
Comment (Betrozov) [inline libs/platform/src/lib/components/table/directives/table-cell-selectable.directive.ts:3]:
```suggestion
import { SPACE } from '@angular/cdk/keycodes';
```

**Review by Betrozov — APPROVED** [review]

**Comment by sKudum** [comment]
Hi @olshansky,

Thank You for the fix. hope remaning wll be covered in future PRs,

Few points in current PR noticiable issues new.

1).On tab focus is not moving to row.
2). On dropdown select in dialog parent screen is displaying the dropdown values.

3).All table are showing blue focus.

**Comment by fkolar** [comment]
> Hi @olshansky,
> 
> Thank You for the fix. hope remaning wll be covered in future PRs,
> 
> Few points in current PR noticiable issues new.
> 
> 1).On tab focus is not moving to row.
> 2). On dropdown select in dialog parent screen is displaying the dropdown values.
> 
> 3).All table are showing blue focus.
> 

@KevinOkamoto @dimamarksman, this is what we discussed today. What is the data representation of this filtering dialog so we can use it in our API.

**Review by InnaAtanasova — APPROVED** [review]

**Comment by olshansky** [comment]
> 1).On tab focus is not moving to row.
> 2). On dropdown select in dialog parent screen is displaying the dropdown values.
> 3).All table are showing blue focus.

@sKudum 
1)&nbsp; Do you mean each particular data row? If yes, let's move it to the "keyboard navigation" feature
1)&nbsp; fixed
1)&nbsp; fixed _(according to fiori design focus outline)_

File: libs/platform/src/lib/components/table/table.component.ts [hunk 2]
```diff
@@ -576,6 +585,18 @@ export class TableComponent<T = any> extends Table implements AfterViewInit, OnD
...
 
+    /** @hidden
+     *  Needs to prevent scrolling and other events on loading.
+     *  TODO: refactor it on keyboard navigation implementation
+     * */
+    @HostListener('keydown', ['$event'])
+    keyDownHandler(event: KeyboardEvent): void {
+        if (this.loading) {
+            event.preventDefault();
+            event.stopPropagation();
```
Comment (dimamarksman) [inline libs/platform/src/lib/components/table/table.component.ts:596]: do we really need stopPropagation?

**Review by sKudum — APPROVED** [review]

**Review by dimamarksman — APPROVED** [review]

**Pull request merged**